docs/src/man/categorical.md

```diff
@@ -2,52 +2,176 @@
 
 Often, we have to deal with factors that take on a small number of levels:
 
-```julia
-v = ["Group A", "Group A", "Group A",
-     "Group B", "Group B", "Group B"]
+```jldoctest
+julia> v = ["Group A", "Group A", "Group A", "Group B", "Group B", "Group B"]
+6-element Array{String,1}:
+ "Group A"
+ "Group A"
+ "Group A"
+ "Group B"
+ "Group B"
+ "Group B"
+
 ```
 
 The naive encoding used in an `Array` represents every entry of this vector as a full string. In contrast, we can represent the data more efficiently by replacing the strings with indices into a small pool of levels. This is what the `CategoricalArray` type does:
 
-```julia
-cv = CategoricalArray(["Group A", "Group A", "Group A",
-                       "Group B", "Group B", "Group B"])
+```jldoctest
+julia> using CategoricalArrays
+
+julia> cv = CategoricalArray(["Group A", "Group A", "Group A",
+                              "Group B", "Group B", "Group B"])
+6-element CategoricalArrays.CategoricalArray{String,1,UInt32}:
+ "Group A"
+ "Group A"
+ "Group A"
+ "Group B"
+ "Group B"
+ "Group B"
+
 ```
 
 `CategoricalArrays` support missing values via the `Nulls` package.
 
-```julia
-using Nulls
-cv = CategoricalArray(["Group A", null, "Group A",
-                       "Group B", "Group B", null])
+```jldoctest
+julia> using CategoricalArrays, Nulls
+
+julia> cv = CategoricalArray(["Group A", null, "Group A",
+                              "Group B", "Group B", null])
+6-element CategoricalArrays.CategoricalArray{Union{Nulls.Null, String},1,UInt32}:
+ "Group A"
+ null
+ "Group A"
+ "Group B"
+ "Group B"
+ null
 ```
 
 In addition to representing repeated data efficiently, the `CategoricalArray` type allows us to determine efficiently the allowed levels of the variable at any time using the `levels` function (note that levels may or may not be actually used in the data):
 
-```julia
-levels(cv)
+```jldoctest
+julia> using CategoricalArrays, Nulls
+
+julia> cv = CategoricalArray(["Group A", null, "Group A",
```

Comment:
If you give a name to the previous test and use the same here, you don't need to repeat these commands.